### PR TITLE
Schema Browser Updates: Routes as Schema

### PR DESCRIPTION
**Pull request by dakota002**

### CODE REVIEW

**Comment by dakota002** [comment]
@lpsinger I added back some UI for selecting the version:

Closed:
image

Open:
image

What are your thoughts?

**Comment by lpsinger** [comment]
> What are your thoughts?

I like it, except that this too feels like an application for the rich dropdown control that @tylerbarna needs.

**Comment by dakota002** [comment]
Do they have a component for that, or need one? I could pop this out into a reusable component. It's mostly just a [details](https://www.w3schools.com/tags/tag_details.asp) tag with some css

**Comment by tylerbarna** [comment]
> Do they have a component for that, or need one? I could pop this out into a reusable component. It's mostly just a [details](https://www.w3schools.com/tags/tag_details.asp) tag with some css

yeah, this would pretty much address the needs for the popup menu, provided it's possible to set its z-index to display over other page elements rather than displacing them and also that it can accept child components

**Comment by dakota002** [comment]
I'll break it out as a separate PR

**Comment by dakota002** [comment]
The component PR: https://github.com/nasa-gcn/gcn.nasa.gov/pull/1147

**Comment by lpsinger** [comment]
Would you please give this PR a more descriptive name?

File: app/routes/docs/schema-browser/latest.$.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  owner: 'nasa-gcn',
+  repo: 'gcn-schema',
+}
+const octokit = new Octokit()
+
+export async function loader({ params: { '*': path } }: DataFunctionArgs) {
+  let latestRelease
+  try {
+    latestRelease = (await octokit.rest.repos.getLatestRelease(githubData)).data
+  } catch (error) {
```
Comment (lpsinger) [inline app/routes/docs/schema-browser/latest.$.ts:15]: What exception does it raise if there have been no releases? Can we make this `catch` block more specific?

File: app/routes/docs/schema-browser/latest.$.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { Octokit } from '@octokit/rest'
+import type { DataFunctionArgs } from '@remix-run/node'
+import { redirect } from '@remix-run/node'
+
+const githubData = {
+  owner: 'nasa-gcn',
+  repo: 'gcn-schema',
+}
+const octokit = new Octokit()
```
Comment (lpsinger) [inline app/routes/docs/schema-browser/latest.$.ts:9]: Reuse the existing octokit client instance in schema-data.ts.

That may mean defining a function in schema-data.ts that gets the latest release.

File: app/routes/docs.tsx [hunk 3]
```diff
@@ -103,7 +103,10 @@ export default function () {
                 </>,
                 ...(enableSchemaBrowser
                   ? [
-                      <NavLink key="schema-browser" to="schema-browser">
+                      <NavLink
+                        key="schema-browser"
+                        to="schema-browser/main/gcn"
```
Comment (lpsinger) [inline app/routes/docs.tsx:108]: This should be a link to the latest release, not the main branch.

File: app/routes/docs/schema-browser/$version/$.tsx [hunk 4]
```diff
@@ -1,34 +1,164 @@
...
+} from '~/lib/schema-data'
 
-  return { path, result, examples }
+export async function loader({
+  params: { version, '*': path },
+}: DataFunctionArgs) {
+  if (!version) throw new Response(null, { status: 404 })
+  if (path?.endsWith('/')) {
+    return redirect(`${path.slice(0, -1)}`)
+  }
```
Comment (lpsinger) [inline app/routes/docs/schema-browser/$version/$.tsx:41]: Why is this necessary?

**Review by lpsinger — CHANGES_REQUESTED** [review]
Move the call to `getVersionRefs()` and the version dropdown component one route module higher so that we don't need to query GitHub for the version refs on every single request.

File: app/routes/docs/schema-browser/$version/$.tsx (hunk 4 above)
Comment (dakota002) [reply app/routes/docs/schema-browser/$version/$.tsx:41]: Ah maybe not totally needed, there were errors when the path ended with '/' because of how it tried to get the content data from the api. We could just let it throw the error

Comment (lpsinger) [reply app/routes/docs/schema-browser/$version/$.tsx:41]: I guess it's fine.

File: app/routes/docs/schema-browser/latest.$.ts (hunk 1 above)
Comment (lpsinger) [reply app/routes/docs/schema-browser/latest.$.ts:15]: I suggest catching 404 errors but re-raising anything else.

Comment (dakota002) [reply app/routes/docs/schema-browser/latest.$.ts:15]: Is there a correct way to implement that with like ` if (error instanceof ???)`  

Or some other way to catch the 404?

Comment (lpsinger) [reply app/routes/docs/schema-browser/latest.$.ts:15]: Something like that, yes.

Comment (dakota002) [reply app/routes/docs/schema-browser/latest.$.ts:15]: Any suggestions? Or examples? There is no class exported by octokit that allows for the instanceof method

Comment (lpsinger) [reply app/routes/docs/schema-browser/latest.$.ts:15]: https://github.com/octokit/octokit.js#request-error-handling

Comment (dakota002) [reply app/routes/docs/schema-browser/latest.$.ts:15]: Perfect, thanks!

**Comment by dakota002** [comment]
I missed the thing about moving the version selector, working on that now

**Comment by dakota002** [comment]
Its ready for review

File: app/routes/docs/schema-browser.tsx [hunk 5]
```diff
@@ -5,127 +5,75 @@
...
+import { getVersionRefs } from '~/lib/schema-data'
 
-// Schema treeItem
-type SchemaTreeItem = {
-  name: string
-  path: string
-  children?: SchemaTreeItem[]
-}
+export async function loader({
+  params: { version, '*': path },
```
Comment (lpsinger) [inline app/routes/docs/schema-browser.tsx:20]: How will this route ever have params? The route module name does not contain any splats.

**Review by lpsinger — CHANGES_REQUESTED** [review]

File: app/routes/docs/schema-browser.tsx (hunk 5 above)
Comment (lpsinger) [reply app/routes/docs/schema-browser.tsx:20]: Interesting.

**Review by lpsinger — APPROVED** [review]
